frontend/src/pages/register.js

```diff
@@ -0,0 +1,212 @@
+// adapted heavily from H4I PI's register workflow examples found here:
+// https://github.com/hack4impact-uiuc/infra-authentication-examples/blob/master/nextjs-client-example/pages/register.js
+
+import { Component } from 'react'
+import Router from 'next/router'
+import { registerUser, loginGoogleUser } from '../utils/api'
+import { GoogleLogin } from 'react-google-login'
+import Nav from '../components/nav'
+import Head from '../components/head'
+import cookie from 'js-cookie'
+import {
+  Container,
+  Form,
+  Button,
+  FormGroup,
+  Label,
+  Input,
+  Card,
+  CardBody,
+  CardTitle,
+  Modal,
+  ModalBody,
+  ModalFooter,
+  ModalHeader
+} from 'reactstrap'
+
+const EMAIL_REGEX =
+  "([a-zA-Z0-9!#$%&'*+/=?^_`{|}~-]+)@([a-zA-Z0-9!#$%&'*+/=?^_`{|}~-]+).([a-zA-Z]{2,3}).?([a-zA-Z]{0,3})"
+
+class RegisterPage extends Component {
+  constructor(props) {
+    super(props)
+    this.state = {
+      email: '',
+      password: '',
+      passwordVerification: '',
+      errorMessage: '',
+      showInvalidPasswordModal: false,
```

Comment:
would probably be better to have 1 variable displaying which type of modal to show: `InvalidPasswordModal`, `InvalidRequestModal`, `InvalidGoogleRequestModal`, or `None`. Unless it's possible to have multiple modals?

Reply:
(you can handle this in a follow-up PR so we can ship this one faster)